aggregator_api/src/lib.rs

```diff
@@ -109,15 +112,55 @@ async fn get_task_ids<C: Clock>(
 async fn post_task<C: Clock>(
     _: &mut Conn,
     (State(ds), Json(req)): (State<Arc<Datastore<C>>>, Json<PostTaskReq>),
-) -> Result<impl Handler, Status> {
-    let vdaf_verify_keys = Vec::from([SecretBytes::new(
-        thread_rng()
-            .sample_iter(Standard)
-            .take(req.vdaf.verify_key_length())
-            .collect(),
-    )]);
+) -> Result<impl Handler, impl Handler> {
+    let task_id = req.task_id.unwrap_or_else(random);
+    let vdaf_verify_keys = if let Some(encoded) = req.vdaf_verify_key.as_ref() {
+        let bytes = URL_SAFE_NO_PAD.decode(encoded).map_err(|err| {
+            (
+                format!("Invalid base64 value for vdaf_verify_key: {err}"),
+                Status::BadRequest,
+            )
```

Comment:
This doesn't work because `<String as Handler>` uses `conn.ok(...)`, which halts the connection. Then, the tuple handler does not run `<Status as Handler>` because the connection is halted. I think I'll set the body with a mutating method instead, and go back to returning `Status` out of the `ApiHandler`.

Reply:
This is a known issue that I'd like to change in the next breaking trillium release